resources/assets/js/src/components/chart/Chart.tsx

```diff
@@ -209,62 +209,49 @@ export default class Chart extends React.Component<IProps, IState> {
 
     private loadData(dateFrom: string, dateTo: string, direction: string = null): void {
-        let url = this.props.url;
+        const url = this.createRequestURL(dateFrom, dateTo);
         let dataMeta;
 
-        if (dateFrom !== null && dateTo !== null) {
-            url = `${url}/?start_date=${dateFrom}&end_date=${dateTo}`;
-        }
-
         axios.get(url).then((response: any) => {
-            const newData = [];
-            const length = this.state.data.length > 0 ? response.data.length - 1 : response.data.length;
-
-            for (let i = 0; i < length; i++) {
-                newData.push({
-                    date: response.data[i].date,
-                    value: response.data[i][this.props.columnName],
-                });
-            }
-
-            if (newData[0].date === this.state.initialDate) {
+            const newData = this.processResponse(response);
+            if (!newData.length || newData[0].date === this.state.initialDate) {
                 return;
             }
 
             if (direction) {
-                const data = [...this.state.data];
-
-                switch (direction) {
-                    case "plus":
-                        data.push(...newData);
-                        break;
-                    case "minus":
-                        data.unshift(...newData);
-                        break;
-                    default:
-                        return null;
-                }
-
+                let data = (direction === "plus") ? [...this.state.data, ...newData] : [...newData, ...this.state.data];
+                data = ArrayUtil.removeDuplicities(data);
                 dataMeta = {
-                    firstDate: newData[0].date,
-                    lastDate: newData[newData.length - 1].date,
+                    firstDate: data[0].date,
+                    lastDate: data[data.length - 1].date,
                 };
-
-                this.setState({
-                    data,
-                    dataMeta,
-                });
+                this.setState({ data, dataMeta });
             } else {
                 dataMeta = {
                     firstDate: newData[0].date,
                     lastDate: newData[newData.length - 1].date,
                 };
-
-                this.setState({
-                    data: newData,
-                    dataMeta,
-                });
+                this.setState({ data: newData, dataMeta });
             }
         });
     }
+
+    private processResponse(response: any): Array<{date: string, value: any}> {
+        return response.data.map((row) => {
+            return {
+                date: row.date,
+                value: row[this.props.columnName],
+            };
+        });
+    }
+
+    private createRequestURL(dateFrom: string, dateTo: string): string {
+        let url = this.props.url;
+
+        if (dateFrom !== null && dateTo !== null) {
+            url = `${url}/?start_date=${dateFrom}&end_date=${dateTo}`;
```

Comment:
It is possible to build URL with an object? Hardwired string are hard to maintain. The same for other hardcoded URLs

Reply:
Might some kind of Util

Reply:
you mean function in DateUtil that generates url based on passed object?